tools/ci_build/github/azure-pipelines/templates/android-qnn-java-api-aar.yml

```diff
@@ -0,0 +1,115 @@
+parameters:
+- name: buildConfig
+  displayName: Configuration to build
+  type: string
+  default: 'Release'
+
+- name: buildSettings
+  displayName: Path to the build setting file
+  type: string
+  default: ''
+
+- name: includedOpsConfig
+  displayName: Path to the included ops config file
+  type: string
+  default: ''
+
+- name: artifactName
+  displayName: Artifact Name
+  type: string
+  default: ''
+
+- name: job_name_suffix
+  displayName: Job name
+  type: string
+  default: ''
+
+- name: publish_executables
+  # The executables will only be published if the publish_executables is set to "1"
+  displayName: Publish executables such as onnxruntime_perf_test, ...
+  type: string
+  default: '0'
+
+- name: pool_name
+  displayName: Pool name
+  type: string
+  default: 'onnxruntime-Ubuntu2204-AMD-CPU'
+
+- name: packageName
+  displayName: Package Name
+  type: string
+  default: 'onnxruntime-android-qnn'
+
+jobs:
+- job: Android_QNN_Java_API_AAR_Packaging_${{ parameters.job_name_suffix }}
+  timeoutInMinutes: 120
+  workspace:
+    clean: all
+  pool: ${{parameters.pool_name}}
+
+  variables:
+    artifacts_directory: $(Build.BinariesDirectory)/.artifacts
+
+  steps:
+  - checkout: self
+    clean: true
+    submodules: none
+
+  - task: CmdLine@2
+    displayName: Create artifacts directory
+    inputs:
+      script: |
+        # Create a folder for artifacts
+        mkdir -p $(artifacts_directory)
+      workingDirectory: $(Build.BinariesDirectory)
+
+  #TODO: use a different docker file since this job doesn't need to rely on manylinux
+  - template: get-docker-image-steps.yml
+    parameters:
+      Dockerfile: tools/ci_build/github/linux/docker/Dockerfile.manylinux2_28_cpu
+      Context: tools/ci_build/github/linux/docker
+      DockerBuildArgs: "--build-arg BUILD_UID=$( id -u )"
+      Repository: onnxruntimecpubuild
+
+  - template: set-version-number-variables-step.yml
+
+  - template: use-android-ndk.yml
+
+  - template: jobs/download_linux_qnn_sdk.yml
```

Comment:
this template seems pretty close to android-java-api-aar.yml. I wonder if it would be possible to reuse that one? maybe we could add an optional parameter for including the QNN SDK and download it if specified.

it's possible to conditionally include steps.
https://learn.microsoft.com/en-us/azure/devops/pipelines/process/template-expressions?view=azure-devops#conditional-insertion